index.md

```diff
@@ -1,63 +1,200 @@
+
+Flutter includes a functional-reactive framework, a 2D rendering engine, ready-
+made widgets, and development tools. The components work together to help you 
+design, build, test, and debug apps. Everything is organized around a few core 
+principles. 
+
+### Everything's a Widget
+
+Widgets are the basic building blocks of every Flutter app. Each widget is an 
+immutable declaration of part of a user interface.  Unlike other frameworks that 
+separate views, controllers, layouts, and other properties, Flutter has a 
+consistent, unified object model: the widget.
+
+A widget can define:
+
+*   a structural element (like a button or menu)
+*   a stylistic element (like a font or color scheme)
+*   an aspect of layout (like padding)
+*   some business logic 
+*   and so on...
+
+Widgets form a hierarchy based on composition.  Each widget nests inside, and 
+inherits properties from, its parent.  There is no separate "application" object.
+Instead, the root widget serves this role.
+
+
+You can respond to events, like user interaction, by telling the framework to 
+replace a widget in the hierarchy with another widget.  The framework then 
+compares the new and old widgets and efficiently updates the user interface.
+
+#### Composition > inheritance
+
+Widgets are themselves often composed of many small, single-purpose widgets that
+combine to produce powerful effects.  For example, [Container](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/container.dart),
+a commonly-used widget, is made up of several widgets responsible for layout,
+painting, positioning, and sizing. Specifically, Container is made up of
+[LimitedBox](https://docs.flutter.io/flutter/widgets/LimitedBox-class.html),
+[ConstrainedBox](https://docs.flutter.io/flutter/widgets/ConstrainedBox-class.html),
+[Align](https://docs.flutter.io/flutter/widgets/Align-class.html),
+[Padding](https://docs.flutter.io/flutter/widgets/Padding-class.html),
+[DecoratedBox](https://docs.flutter.io/flutter/widgets/DecoratedBox-class.html),
+and [Transform](https://docs.flutter.io/flutter/widgets/Transform-class.html)
+widgets.  Rather than subclassing Container to produce a customized effect, you
+can compose these, and other, simple widgets in novel ways.
+
+The class hierarchy is shallow and broad to maximize the possible number of 
+combinations.
+
+![Widget class hierarchy](/images/whatisflutter/Widget_class_hierarchy.png){:style="max-width: 900px"}
+
+You can also control the *layout* of a widget by composing it with other widgets.
+For example, to center a widget, you wrap it in a Center widget. There are
+widgets for padding, alignment, row, columns, and grids. These layout widgets
+do not show themselves in an app's UI.  Instead, their sole purpose is to
+control some aspect of another widget's layout. To understand why a widget
+renders in a certain way, it's often helpful to inspect the neighboring widgets.
+
+#### Layer cakes are fun and functional
```

Comment:
Sorry, lost my nerve there and made it boring. Changed back to delicious! (Could also change to "good for you"...)